portfolio/src/main/webapp/script.js

```diff
@@ -117,40 +117,73 @@ const sleep = async (time_ms) => (
  *    from the server.
  */
 const loadComments = async (maxComments) => {
+  /*
+   * The /comments route exposes a simple API.
+   * This API is available only using the GET http method.
+   * Requests sent to this API must have a body which contains:
+   *   - nothing
+   *   - A single parameter, 'maxComments'
+   * If the request body is non-empty, then maxComments is used to limit
+   * the number of results returned by the request.
+   * The response body will be encoded as json.
+   * The response body will contain a single top-level array, whose
+   * elements will all be Comment objects.
+   * Comment objects have three members:
+   *   id: the id of the comment in the server's datastore
+   *   title: the comment title
+   *   text: the comment text
+   */
   const data = await fetch(`/comments?maxComments=${maxComments}`);
```

Comment:
What happens if the fetch fails? Are the comments simply not loaded?

Reply:
At the moment, the failure is ignored and the page loads without comments. It won't negatively impact the aesthetics, since the structural elements related to the comments aren't pushed onto the DOM unless the server responds.

Reply:
Well, you could alert the user that the comments failed to load but that's up to you.